### PR TITLE
Feature/pz 21 implemenation of footer section

### PR DESCRIPTION
**Pull request by Prashishgt**
# Merge Request

## Summary of change

image
image

## Mention

Review: @dipeshkumarsah98 @subhashishjungshah @sarojghalan 

Fixes: PZ-21 Implementation of Footer Section

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that would cause existing functionality to not work as expected)
- [ ] Front End (UI) changes
- [ ] Changes in `documentation`

## How did you test

- [X] Rendered in page.tsx in app for check.

## Checklist

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [X] My code follows the style guidelines of this project
- [ ] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [ ] I have added required tests that prove my fix is effective or that my feature works
- [ ] Both New and Existing unit tests pass locally with my changes

### CODE REVIEW

File: app/page.tsx [hunk 1]
```diff
@@ -1,4 +1,5 @@
 "use client";
+import Footer from "@/components/Footer";
```
Comment (subhashishjungshah) [inline app/page.tsx:2]: naming should be PZFooter ivey

File: components/Footer/index.tsx [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  StyledFooter,
+  StyledFooterLinks,
+  StyledFooterNewsLetter,
+  StyledFooterNewsLetterInput,
+  StyledRightIconButton,
+  StyledFooterNewsLetterWrapper,
+} from "./style";
+
+const FooterLinks = () => (
+  <StyledFooterLinks>
```
Comment (subhashishjungshah) [inline components/Footer/index.tsx:15]: lets make this thing dynamic. Lets create a data folder and add this data inside that. Import and map through the data

File: components/Footer/style.ts [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    width: 100%;
+    gap: 35px;
+    padding-left: 20px;
+  }
+
+  a {
+    text-decoration: none;
+    font-size: 16px;
+    font-weight: 300;
+    color: ${({ theme }) => theme.color.fifthNavLink};
```
Comment (subhashishjungshah) [inline components/Footer/style.ts:23]: what is this naming haha? naming should be meaningful

File: components/theme.ts [hunk 4]
```diff
@@ -4,6 +4,7 @@ export const PZtheme = {
     secondaryColor: "#000",
     tertiaryColor: "#f7f7f7",
     fourthColor: "#e5e5e5",
+    fifthNavLink: "#1c4334c7",
```
Comment (subhashishjungshah) [inline components/theme.ts:7]: if this color is only at one place, you can directly write the color code.

**Review by subhashishjungshah — CHANGES_REQUESTED** [review]

File: components/Footer/index.tsx [hunk 5]
```diff
@@ -0,0 +1,51 @@
+import Link from "next/link";
+import React from "react";
```
Comment (sarojghalan) [inline components/Footer/index.tsx:2]: React is unnecessary here.

File: components/Footer/index.tsx [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+    <Link href="/">
+      <PZTypography variant="footerLink">SHIPPING & RETURN</PZTypography>
+    </Link>
+    <Link href="/">
+      <PZTypography variant="footerLink">SOCIAL</PZTypography>
+    </Link>
+  </StyledFooterLinks>
+);
+
+const FooterNewsLetter = () => (
```
Comment (sarojghalan) [inline components/Footer/index.tsx:31]: naming should be PZFooterNewsLetter

File: components/Footer/index.tsx [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+  <StyledFooterNewsLetter>
+    <StyledFooterNewsLetterInput placeholder="Enter Email" />
+    <StyledRightIconButton>
+      <ArrowRightAltIcon />
+    </StyledRightIconButton>
+  </StyledFooterNewsLetter>
+);
+
+const Footer = () => (
```
Comment (sarojghalan) [inline components/Footer/index.tsx:40]: naming should be PZFooter.

File: components/Footer/style.ts [hunk 8]
```diff
@@ -0,0 +1,82 @@
+import { IconButton, Stack } from "@mui/material";
+import styled from "styled-components";
+
+export const StyledFooterLinks = styled(Stack)`
```
Comment (sarojghalan) [inline components/Footer/style.ts:4]: while creating styled component, naming should be StyledPZFooterLink. Please maintain these standard while creating styled component. And while creating component, component naming should start with PZ.

File: components/Footer/style.ts [hunk 9]
```diff
@@ -0,0 +1,82 @@
+import { IconButton, Stack } from "@mui/material";
```
Comment (sarojghalan) [inline components/Footer/style.ts:1]: since we are using styled component, "use client" flag is missing?

**Review by sarojghalan — CHANGES_REQUESTED** [review]
Nice keep up the pace.

File: components/PZFooter/index.tsx [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+  StyledPZFooterNewsLetter,
+  StyledPZFooterNewsLetterInput,
+  StyledPZRightIconButton,
+  StyledPZFooterNewsLetterWrapper,
+} from "./style";
+
+const PZFooterLinks = () => (
+  <StyledPZFooterLinks>
+    {footerLinks.map((link, index) => (
+      <Link key={index} href={link.href}>
```
Comment (subhashishjungshah) [inline components/PZFooter/index.tsx:18]: Key should never be index

**Pull request merged**